packages/server/src/threads/automation.ts

```diff
@@ -495,6 +495,32 @@ export function execute(job: Job, callback: WorkerCallback) {
   })
 }
 
+export function executeSynchronously(job: Job) {
+  const appId = job.data.event.appId
+  if (!appId) {
+    throw new Error("Unable to execute, event doesn't contain app ID.")
+  }
+
+  const timeoutPromise = new Promise((resolve, reject) => {
+    setTimeout(() => {
+      reject(new Error("Timeout exceeded"))
+    }, job.data.event.timeout || 12000)
+  })
+
+  return context.doInAppContext(appId, async () => {
+    const envVars = await sdkUtils.getEnvironmentVariables()
+    // put into automation thread for whole context
+    return context.doInEnvironmentContext(envVars, async () => {
+      const automationOrchestrator = new Orchestrator(job)
```

Comment:
@mike12345567 I've split the execute synchronously out into a new function and used promise.race here - I know it was slightly different than the flag within execute that we discussed but i thought this kind of ensured we separated sync code from async code. Maybe it makes more sense to it discussed way? I did test and this works well and throws back the `Timeout exceeded` error to the client correctly etc

Reply:
Yes - this is probably acceptable, although flagging to the orchestrator in someway, like through a `cancel` function, just to make sure the orchestrator is aware that it doesn't need to continue.

The concern with this approach is that the sync operation will simply continue, and the response will be ignored, which uses up CPU time. Its not very major, but making the orchestrator aware that the response is no longer needed means we could exit early.